### DISCUSSION TITLE
Milvus Cluster Installation with Helm

### DISCUSSION THREAD

**Discussion by mjafar84** [milvus-io/milvus#22072] [Q&A and General discussion] [answered]
I want to Install Milvus Cluster with Helm, not with minikube for test. As we know we have following components for milvus cluster: Proxy,Root coord,Query coord,Query node,Index coord,Index node,Data coord,Data node,etcd,MinIO,Pulsar. For example I want to install cluster on K8s with 2 master and 3 worker nodeS. Some questions.

  1.  Which components exactly needs volumes for data storage or temporary data?
  2.  Which components recommended/can for scale up to gain performance?
  3.  Is there any support for shared disk with 2 worker nodes and for example minio service work from 2 nodes?
  4.  How can we replace MinIO service with S3? Is it recommended to use S3 instead of MinIO?
  5.  Is there any further recommendation for cluster setup on K8s?

**Comment by yhmo** [comment, marked as answer]
1. Milvus native components: Proxy,Root coord,Query coord,Query node,Index coord,Index node,Data coord,Data node
    Milvus cluster dependencies: Etcd,MinIO(or S3),Pulsar
    
    Etcd stores metadata such as collection scheam, index information, filepath, etc.
    MinIO stores data file such as index file and raw data file.
    Pulsar is like a WAL(write ahead log), all the insert/delete operations will be stored in Pulsar before execution.
    
    In Milvus cluster, only the Etcd/MinIO/Pulsar need volumes for storage.

2. QueryNodes/QueryCoord serve for search, QueryCoord manages QueryNodes, each cluster only has one QueryCoord. You can increase query nodes to gain performance.

3. All the native components share one storage service(S3 or MinIO), share one Pulsar service, share one Etcd service.

4. Configure S3 for milvus: https://milvus.io/docs/deploy_s3.md#Configure-S3-on-K8s
5. Helm is recommended, just follow the install guideline: https://milvus.io/docs/install_cluster-helm.md

**Reply by mjafar84** [reply]
Thanks @yhmo
